snapcraft/internal/errors.py

```diff
@@ -185,6 +185,18 @@ class SnapcraftPartMissingError(SnapcraftError):
     )
 
 
+class NoSuchFileError(SnapcraftError):
```

Comment:
the error message seems to specific for this class' name. How about `NoSuchFileInPartError`.
These messages as with ToolError are getting a bit too un-namespaced btw. We should fix that.